pastas/plotting/bokeh.py

```diff
@@ -0,0 +1,300 @@
+            legend_label="Residuals",
+        )
+        res_plot.line(
+            "index",
+            "Residuals",
+            source=source,
+            color="black",
+            alpha=0.7,
+            legend_label="Residuals",
+        )
+
+        if self._model.settings["noise"]:
+            res_plot.line("index", "Noise", source=source, legend_label="Noise")
+            res_plot.scatter("index", "Noise", source=source, legend_label="Noise")
+
+        res_plot.legend.ncols = 2
+
+        # Parameter Table
+        df = ColumnDataSource(self._model.parameters.loc[:, ["optimal"]])
+        columns = [
+            TableColumn(
+                field="index",
+                title="Name",
+                formatter=StringFormatter(font_style="bold"),
+            ),
+            TableColumn(
+                field="optimal",
+                title="Optimal",
+                formatter=ScientificFormatter(precision=2),
+            ),
+        ]
+        table = DataTable(
+            source=df,
+            columns=columns,
+            editable=False,
+            index_position=None,
+            width=int(0.25 * width),
+            height=heights[0] + heights[2] - 10,
+        )
+
+        left_column = [p, res_plot]
+        right_column = [table]
+
+        # Contributions
+        rfunc_plot = None
+
+        for i, smname in enumerate(self._model.stressmodels.keys(), start=2):
+            if i == int(len(self._model.stressmodels) + 1):
+                x_axis_location = "below"
+            else:
+                x_axis_location = None
+
+            contrib_plot = figure(
+                y_axis_label="Rise",
+                toolbar_location=None,
+                tools=TOOLS,
+                x_axis_location=x_axis_location,
+                width=int(0.75 * width),
+                height=heights[i],
+                x_axis_type="datetime",
+                x_range=p.x_range,
+            )
+
+            if rfunc_plot is not None:
+                xrange = rfunc_plot.x_range
+            else:
+                xrange = False
+
+            rfunc_plot = figure(
+                x_axis_label=None,
+                toolbar_location=None,
+                x_axis_location=x_axis_location,
+                width=int(0.25 * width),
+                height=heights[i],
+            )
+
+            contrib_plot.line("index", smname, source=source)
+            response = self._model.get_step_response(smname)
+            rfunc_plot.line(response.index, response.values)
+
+            left_column.append(contrib_plot)
+            right_column.append(rfunc_plot)
+
+        select = figure(
+            height=50,
+            width=int(0.75 * width),
+            y_range=p.y_range,
+            x_axis_type="datetime",
+            y_axis_type=None,
+            tools="",
+            toolbar_location=None,
+            background_fill_color="#efefef",
+        )
+
+        range_tool = RangeTool(x_range=p.x_range)
```

Comment:
Very cool :)

Reply:
Hm, should add that to the plotly plot as well maybe :innocent: 
https://plotly.com/python/range-slider/

Reply:
I'm still exploring, many fun options! :)